bluetooth: Classic: GOEP: OBEX improvement and bugfix

CanWang001 opened the pull request:
This PR includes some OBEX improvement and bugfix.
Please see the commit for more details.

CanWang001 committed:
Bluetooth: OBEX: add bt_obex_has_app_param helper function

Add a new helper function bt_obex_has_app_param() to check whether
a specific application parameter tag exists in the OBEX buffer.

The function uses the existing bt_obex_get_header_app_param() to
retrieve the application parameters, then parses them using
bt_obex_tlv_parse() with a callback to search for the specified tag ID.

This simplifies checking for the presence of application parameters
without needing to manually parse the TLV structure.

Signed-off-by: Can Wang <can.wang@nxp.com>

CanWang001 committed:
Bluetooth: OBEX: allow zero-length body in End-of-Body header

According to OBEX Version 1.5, section 2.2.9, it is legal to send a
zero-length End-of-Body header when object body data is generated on
the fly and the end cannot be anticipated.

Update parameter validation in bt_obex_add_header_body_or_end_body()
and bt_obex_add_header_end_body() to allow NULL body pointer when
length is zero, while still rejecting NULL body with non-zero length.

This enables proper support for zero-length End-of-Body headers as
specified in the OBEX specification.

Signed-off-by: Can Wang <can.wang@nxp.com>

CanWang001 committed:
Bluetooth: OBEX: fix incorrect error code in obex_transport_disconn

The function obex_transport_disconn() was always returning -EINVAL
regardless of the actual error code returned by the transport
disconnect operation.

Fix this by returning the actual error code from the transport
disconnect call instead of the hardcoded -EINVAL value.

Signed-off-by: Can Wang <can.wang@nxp.com>

lylezhu2012 commented on subsys/bluetooth/host/classic/obex.c:
```diff
@@ -2753,7 +2753,13 @@ int bt_obex_add_header_end_body(struct net_buf *buf, uint16_t len, const uint8_t
+	 * In some cases, the object body data is generated on the fly and the end cannot
+	 * be anticipated, so it is legal to send a zero length End-of-Body header.
+	 */
+	if (!buf || (len && !body)) {
```
```suggestion
	if ((buf == NULL) || ((len != 0) && (body == NULL))) {
```

CanWang001 commented on subsys/bluetooth/host/classic/obex.c:
```diff
@@ -3643,6 +3643,49 @@ int bt_obex_get_header_app_param(struct net_buf *buf, uint16_t *len, const uint8
+{
+	struct bt_obex_has_app_param *data;
+
+	data = user_data;
```
Fixed

CanWang001 committed:
Bluetooth: OBEX: add setpath flags enum

Add documentation for the setpath flags parameter by defining a new
enum bt_obex_setpath_flags that describes the available flags:
- BT_OBEX_SETPATH_FLAG_BACKUP: backup a level before applying name
- BT_OBEX_SETPATH_FLAG_NO_CREATE: don't create folder if it doesn't exist

Update the documentation for bt_obex_setpath() and the setpath callback
in bt_obex_server_ops to reference the new enum, improving clarity for
users of the OBEX setpath operation.

Signed-off-by: Can Wang <can.wang@nxp.com>
